language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -643,14 +643,14 @@ instance ToJSON PackageJson where
 
 -- Read the provided 'package.json'; transform it to include the
 -- provided workspaces; write it back to disk.
-writeTopLevelPackageJson :: FilePath -> [(String, String, [Dependency])] -> FilePath -> IO ()
+writeTopLevelPackageJson :: FilePath -> [(T.Text, T.Text, [Dependency])] -> FilePath -> IO ()
 writeTopLevelPackageJson optOutputDir dependencies file = do
   let (g, nodeFromVertex) = graphFromEdges'
-        (map (\(a, b, ds) -> (a, b, map undependency ds)) dependencies)
+        (map (\(a, b, ds) -> (T.unpack a, T.unpack b, map undependency ds)) dependencies)
```

Comment:
Why do we need to unpack here? Can't the vertex names also be `Text`s?

Reply:
Yes!  So, changing this
```
newtype Dependency = Dependency {undependency :: T.Text}  deriving (Eq, Ord)
```
enables us to lose more pack/unpacks and recover this:
```
        (map (\(a, b, ds) -> (a, b, map undependency ds)) dependencies)
```